File: 프로그래머스/가장 큰 수.py
def solution(numbers):
    strList = list(map(str, numbers))
    strList.sort(key = lambda x : x * 3, reverse = True)
    # int로 한번 바꾸고 다시 str하는 이유는 "0000"을 "0"으로 변환해주기 위함
    return str(int("".join(strList)))

File: 프로그래머스/가장 큰 수.java
import java.util.Arrays;

class Solution {
    public String solution(int[] numbers) {
        String[] arr = Arrays.stream(numbers).boxed().map(String::valueOf).toArray(String[]::new);
        Arrays.sort(arr, (o1, o2) -> {
            return (o2 + o1).compareTo(o1 + o2);
        });
        StringBuilder sb = new StringBuilder();
        for (String a : arr)
            sb.append(a);
        return (sb.toString().replaceAll("0", "").isEmpty()) ? "0" : sb.toString();
    }
}